modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java

```diff
@@ -130,10 +130,10 @@ public void processOpts() {
     @SuppressWarnings({ "static-method", "unchecked" })
     public Map<String, Object> postProcessAllModels(Map<String, Object> objs) {
         if (supportsInheritance) {
-            // Index all CodegenModels by name.
+            // Index all CodegenModels by model name.
             Map<String, CodegenModel> allModels = new HashMap<String, CodegenModel>();
             for (Entry<String, Object> entry : objs.entrySet()) {
-                String modelName = entry.getKey();
+                String modelName = toModelName(entry.getKey());
```

Comment:
Yes, Typescript and PHP set supportsInheritance to true but they don't know anything about the two new `CodegenModel.parentModel` and `CodegenModel.interfaceModel` fields I added, and this change is purely to ensure that those two fields are set correctly. Without this change (to the code I've provided), the will be null for referenced parent and interface definitions with lowerCase model names.